### PR TITLE
Allowed to upload `.xml` and `.json` annotation files from UI

### LINKED ISSUE #5274: Error: File must be file instance with ZIP extension
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour
Expected to be able to upload custom annotations that aren't necessarily  `.zip`, especially when the signature for adding an annotation import allows for different extensions.

`@importer(name="MyFormat", version="2.0", ext="XML")`

### Current Behaviour
Throws `Error: File must be file instance with ZIP extension` when trying to upload.

### Possible Solution
It seems to be a validation bug on the UI side. Can be fixed by removing validation step `cvat-core/src/annotations.ts:line 312`

`throw new ArgumentError('File must be file instance with ZIP extension');`

I may well be missing something in the docs here, but even if I am, I'd like to just be able to upload a single json file that contains annotations in a custom format.

### PR DESCRIPTION
**Pull request by klakhov**
### Motivation and context

Resolved #5274
+ Fixed typo in tests

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by klakhov** [comment]
@bsekachev  Could you look on this pr?

File: cvat-core/src/annotations.ts [hunk 1]
```diff
@@ -305,11 +305,19 @@ export function importDataset(
...
-    if (typeof file === 'string' && !file.toLowerCase().endsWith('.zip')) {
-        throw new ArgumentError('File must be file instance with ZIP extension');
-    }
-    if (file instanceof File && !(['application/zip', 'application/x-zip-compressed'].includes(file.type))) {
-        throw new ArgumentError('File must be file instance with ZIP extension');
+    const allowedFileExtensions = [
+        '.zip', '.xml', '.json',
+    ];
+    if (typeof file === 'string' && !(allowedFileExtensions.some((ext) => file.toLowerCase().endsWith(ext)))) {
+        throw new ArgumentError('File must be file instance with eigther ZIP, XML or JSON extension');
```
Comment (bsekachev) [inline cvat-core/src/annotations.ts:312]: Maybe to use ``allowedFileExtensions`` list in this message?

File: cvat-core/src/annotations.ts [hunk 2]
```diff
@@ -305,11 +305,19 @@ export function importDataset(
...
+    }
+    const allowedMimeTypes = [
+        'application/zip', 'application/x-zip-compressed',
+        'application/xml', 'text/xml',
+        'application/json',
+    ];
+    if (file instanceof File && !(allowedMimeTypes.includes(file.type))) {
+        throw new ArgumentError('File must be file instance with eigther ZIP, XML or JSON extension');
```
Comment (bsekachev) [inline cvat-core/src/annotations.ts:320]: The same comment

**Comment by bsekachev** [comment]
Please, look at eslint issues

**Comment by bsekachev** [comment]
okay, issues related with core part, I will merge it as is.

**Pull request merged**